geode-core/src/main/java/org/apache/geode/internal/tcp/ConnectionTable.java

```diff
@@ -494,20 +494,28 @@ Connection getThreadOwnedConnection(DistributedMember id, long startTime, long a
 
     // Update the list of connections owned by this thread....
 
-    if (this.threadConnectionMap == null) {
+    ConcurrentMap cm = this.threadConnectionMap;
+    if (cm == null) {
       // This instance is being destroyed; fail the operation
       closeCon(
           "Connection table being destroyed",
           result);
       return null;
     }
 
-    ArrayList al = (ArrayList) this.threadConnectionMap.get(id);
+    ArrayList al = (ArrayList) cm.get(id);
     if (al == null) {
       // First connection for this DistributedMember. Make sure list for this
       // stub is created if it isn't already there.
       al = new ArrayList();
 
+      if (this.threadConnectionMap == null) {
```

Comment:
Thanks for comments! Can we just do `synchronized(this.threadConnectionMap)` at the start and not use copy to local variable? In this case we don't need always to check if threadConnectionMap is changed as it can be only changed by this thread.

Reply:
I'm not up to speed on this code but I see close() just nulling out this.threadConnectionMap. It does not actually iterator over its contents and close the cons in it. I think this might be happening in another structure: this.threadConnMaps. I would consider changing the code to not null out this.threadConnectionMap. Then you could make this field final and get rid of all the null checks. If we do need to test for it being null then can you use this.closed instead?